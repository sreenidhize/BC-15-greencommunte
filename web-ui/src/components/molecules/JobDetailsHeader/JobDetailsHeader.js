import React, {useState} from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import CompanyLogo from '../../atoms/CompanyLogo/CompanyLogo';
import CardLabel from '../../atoms/cardLabel/CardLabel';
import Button from '../../atoms/button/Button';
import {makeStyles, Snackbar} from "@material-ui/core";
import { apis } from '../../../utils/resources/resources';
import {Alert} from "@material-ui/lab";
import { headerInfo, jobSavedSuccess, jobAlreadyPresent, jobSavedFailed, failedToRemove } from '../../../utils/constants/constant';

const useStyles = makeStyles({
    buttonsStyle: {
        width: "220px",
        marginTop: "12px",
    },
    columnMargin :{
        marginLeft:"8px",
    },
    root:{
        display: "flex"
    }
});

const JobDetailsHeader = ({ url, alt, jobId, jobTitle, companyName, location, contentDescription, onClickApply, isPrimary, showFindJobs, ...props }) => {
    const classes = useStyles();
    const saveOrRemoveButton = showFindJobs ? "Save" : "Remove";
    const [message, setMessage] = useState("Message");
    const jobsUrl = new URL(apis.GET_SAVEDJOBS + jobId);
    const [messageType, setMessageType] = useState("success");
    const savedJobsUrl = new URL(apis.GET_SAVEDJOBS + jobId);
    const [showMessage, setShowMessage] = useState(false);
    const onClickSaveOrRemove = () => {
        if(saveOrRemoveButton === "Save") {
            fetch(jobsUrl, { 
                method: 'POST', 
                headers: {
                    "Content-type": headerInfo
                } 
            }) 
            .then((res) => {
                if (res.status === 200){
                    setMessage(jobSavedSuccess);
                    setMessageType("success");
                } else if (res.status === 409){
                    setMessageType("info");
                    setMessage(jobAlreadyPresent);
                } else {
                    setMessage(jobSavedFailed);
                    setMessageType("error");
                }
                setShowMessage(true);
            });
        }
        else {
            fetch(savedJobsUrl, {
            method: 'DELETE'})
            .then(res => {
                if (res.status === 200){
                    props.itemsRemoved(jobId);
                } else {
                    setMessage(failedToRemove);
                    setMessageType("error");
                    setShowMessage(true);
                }
                setShowMessage(true);
            });

        }
    };
    return (
        <Grid className={classes.root} data-testid="jobDetailsHeader-placeholder">
            <CompanyLogo url={url} alt={alt} ></CompanyLogo>

            <Grid container direction="column" className={classes.columnMargin}>
                <CardLabel label={jobTitle} isPrimary> </CardLabel>
                <CardLabel label={companyName}> </CardLabel>
                <CardLabel label={location}> </CardLabel>
                { isPrimary ?
                <Grid container justify="space-between" className={classes.buttonsStyle} >
                    <Button label={saveOrRemoveButton} type="small" variant="outlined" onClick={onClickSaveOrRemove} />
                    <Button label="Apply" variant="contained" type="small" onClick={onClickApply} />
                </Grid> :
                ""
                }
            </Grid>
            <Snackbar
                anchorOrigin={{vertical: 'top', horizontal: 'center'}}
                open={showMessage}
                autoHideDuration={3000}
                onClose={()=> {setShowMessage(false);}}
            >
                <Alert severity={messageType} elevation={6} variant="filled">
                    {message}
                </Alert>
            </Snackbar>
        </Grid>

    );
};

JobDetailsHeader.propTypes = {
    url: PropTypes.string,
    alt: PropTypes.string,
    jobTitle: PropTypes.string,
    companyName: PropTypes.string,
    location: PropTypes.string,
    isPrimary: PropTypes.bool,
};

export default JobDetailsHeader;