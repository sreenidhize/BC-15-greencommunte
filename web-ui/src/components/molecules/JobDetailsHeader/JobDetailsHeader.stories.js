import React from 'react';
import JobDetailsHeader from './JobDetailsHeader';
import theme from '../../../utils/theme/theme';
import { ThemeProvider } from '@material-ui/core';

export default {
    title: 'molecules/CardDetailsHeader',
    component: JobDetailsHeader,
};


const Template = (args) => <ThemeProvider theme={theme}><JobDetailsHeader {...args} /></ThemeProvider>;

export const Description = Template.bind({});
Description.args = {
    url: "https://m.media-amazon.com/images/I/61b2FkbZsEL._AC_SS350_.jpg",
    alt: "BMW logo",
    jobTitle: "User Experience Designer",
    companyName: "BMW",
    location:"Hitech City, Telangana",
    isPrimary: true,
};
