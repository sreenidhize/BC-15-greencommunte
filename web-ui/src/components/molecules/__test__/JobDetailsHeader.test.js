import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import JobDetailsHeader from '../JobDetailsHeader/JobDetailsHeader';


global.fetch = jest.fn(() =>
  Promise.resolve()
);

describe("JobDetailsHeader placeholder test", () => {
  it("Matches the snapshot", () => {
    const { container } = render(
      <JobDetailsHeader  />
    );
    expect(container).toMatchSnapshot();
  });

  test("test JobDetailsHeader render", () => {
    
    const { getByTestId } = render(<JobDetailsHeader />);
    expect(getByTestId("jobDetailsHeader-placeholder")).toBeInTheDocument();
  });
});