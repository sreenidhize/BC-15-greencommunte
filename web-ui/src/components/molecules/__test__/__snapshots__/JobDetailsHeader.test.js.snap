// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDetailsHeader placeholder test Matches the snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-3"
    data-testid="jobDetailsHeader-placeholder"
  >
    <div
      data-testid="img-placeholder"
    >
      <img
        class="makeStyles-secondary-5"
      />
    </div>
    <div
      class="MuiGrid-root makeStyles-columnMargin-2 MuiGrid-container MuiGrid-direction-xs-column"
    >
      <h4
        class="MuiTypography-root makeStyles-primary-6 MuiTypography-h4"
        data-testid="cardLabel-placeholder"
      />
      <h4
        class="MuiTypography-root makeStyles-secondary-7 MuiTypography-h4"
        data-testid="cardLabel-placeholder"
      />
      <h4
        class="MuiTypography-root makeStyles-secondary-7 MuiTypography-h4"
        data-testid="cardLabel-placeholder"
      />
      
    </div>
  </div>
</div>
`;
